### ISSUE TITLE
Please create an AppData file

### ISSUE THREAD

**Issue by marekr** [marekr/kicad-issue-test#496]
Please consider writing and installing an AppData files with the application descriptions and some screenshots, else KiCad and the other applications shipped with KiCad look really bad in the GNOME and KDE Software Centers. We'd love to showcase more applications (especially for electronic design), but without the extra data file we can't. See http://people.freedesktop.org/~hughsient/appdata/ for details; thanks!

Richard

**Comment by marekr** [comment]
(by nickoe)
This is a great suggestion, but I am not much for following a "standard" or format where the examples do not validate with the tools to do so.

Testing with the one online on the linked site I get:
Validation failed!

kicad.appdata.xml did not validate:

    [screenshot] url not found
    Not enough [p] content before [ul]
    [li] is too short

And when I try:
appdata-validate kicad.appdata.xml 
kicad.appdata.xml 1 problems detected:
• markup invalid        : start tag <metadata_license> not allowed fappdata-validate kicad.appdata.xml 
kicad.appdata.xml 1 problems detected:
• markup invalid        : start tag <metadata_license> not allowed from section <application> [ln:5 ch:21]rom section <application> [ln:5 ch:21]

As far as I can see the metadata license is stated as "CC0-1.0" and also listed on the reffered spdx site; http://spdx.org/licenses/.

Basically I see two different validators giving false problems. I might want to create the file, but you are also welcome. (I have not commit access, but the best chance of inclusion in kicad is to attach a suggestion here.)

**Comment by marekr** [comment]
(by richard-hughes)
Hi -- Where is kicad.appdata.xml linked from? Also, you want a newer version of appdata-tools -- metadata_license and the SPDX requirement were added in the last 6 months.

**Comment by marekr** [comment]
(by nickoe)
That file was the example file from the linked website. I intended to write it. I used the validator in the arch repo, i don't know if that is out of date.

**Comment by marekr** [comment]
(by nickoe)
Is appdata-tools 0.1.7 not the latest?

**Comment by marekr** [comment]
(by richard-hughes)
No, 0.1.8

**Comment by marekr** [comment]
(by stambaughw)
Has there been any progress on this?  If not, I an willing try creating an AppData file.  I think it would be useful for the project to include this for Linux packaging.  I may need to enlist some help to get the proper screenshots on kicad-pcb.org.

**Comment by marekr** [comment]
(by nickoe)
i was intending to write a proposal, but I was demotivated by validators not giving the same results (mainly because mine was one version number off). So no duplicate effort is done if you take the task Wayne.

I am not sure about the webpage, but yes that was also a thing I was thingking about. I guess the URL do not need to be nice, just static.

**Comment by marekr** [comment]
(by mangelajo)
May be we could just host the .xml + images in a folder inside the project bzr, and getting those copied to the static web server (dev.kicad-pcb.org) when CI triggers.


This way we get two things at once (the xml & images) are kept at a safe place, developers can update it, packagers can fetch it..., and we can automatically publish the images over http..

**Comment by marekr** [comment]
(by stambaughw)
On 6/5/2014 6:28 AM, Nick Østergaard wrote:
> That file was the example file from the linked website. I intended to
> write it. I used the validator in the arch repo, i don't know if that is
> out of date.
> 

Nick,

If you don't mind, please send me what you have done so far and I will
take it from there.  I'll give you author credit for the commit.  Thank
you for taking the initiative on this.

Wayne

**Comment by marekr** [comment]
(by stambaughw)
On 8/26/2014 2:49 PM, Miguel Angel Ajo wrote:
> May be we could just host the .xml + images in a folder inside the
> project bzr, and getting those copied to the static web server (dev
> .kicad-pcb.org) when CI triggers.
> 
> 
> This way we get two things at once (the xml & images) are kept at a safe place, developers can update it, packagers can fetch it..., and we can automatically publish the images over http..
> 

Can you put the images on dev.kicad-pcb.org?  I do not have write access
to make the changes myself.

**Comment by marekr** [comment]
(by nickoe)
2014-08-30 1:52 GMT+02:00 Wayne Stambaugh <email address hidden>:
> On 6/5/2014 6:28 AM, Nick Østergaard wrote:
>> That file was the example file from the linked website. I intended to
>> write it. I used the validator in the arch repo, i don't know if that is
>> out of date.
>>
>
> Nick,
>
> If you don't mind, please send me what you have done so far and I will
> take it from there.  I'll give you author credit for the commit.  Thank
> you for taking the initiative on this.
>
> Wayne

Wayne,

I hereby attach my attempt to write the an kicad.appdata.xml. I am not
sure what the mandatory updatecontact field should be. The copyright
comment is not need I think.

The screenshot links should probably be updated. The appdata page says
"Don't add shadows to your screenshots.", which is on the screenshots
on the webpage currently.

Nick
